src/main/java/com/example/foode/Offer/OfferServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.example.foode.Offer;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class OfferServiceImpl implements OfferService {
+
+    private final OfferRepository offerRepository;
+
+    @Override
+    public Offer createOffer(Offer offer) {
+        return offerRepository.saveAndFlush(offer);
+    }
+
+    @Override
+    public Page<Offer> getOffersByCity(Long cityId, Pageable pageable) {
+        return offerRepository.findAllByCityId(cityId, pageable);
+    }
+
+    @Override
+    public Optional<Offer> getOffer(Long id) {
+        return offerRepository.findById(id);
+    }
+
+    @Override
+    public void deleteOffer(Long id) {
+        offerRepository.deleteById(id);
+    }
+
+    @Override
+    public Offer updateOffer(Offer newOffer, Long id) {
+        return offerRepository.findById(id)
+                .map(offer -> {
+                    offer.setAvailability(newOffer.getAvailability());
+                    offer.setCity(newOffer.getCity());
+                    offer.setDate(newOffer.getDate());
+                    offer.setDescription(newOffer.getDescription());
+                    offer.setPrice(newOffer.getPrice());
+                    offer.setProduct(newOffer.getProduct());
+                    offer.setCity(newOffer.getCity());
+                    return offerRepository.saveAndFlush(offer);
```

Comment:
bro after seeing this, my eyes jumped off the bridge and then shot each other in the head
1. Method in map shouldnt be nested -> if you need to do something bigger than one line you should write private method for that and use it in map
2. You should map this like this -> i see 2 better option
a) use objectMapper
b) add static factory method in offer class (probably sth like from(Offer offer){}